addons/product/models/product_product.py

```diff
@@ -304,9 +304,13 @@ def _compute_partner_ref(self):
 
     def _compute_variant_item_count(self):
         for product in self:
-            domain = ['|',
-                '&', ('product_tmpl_id', '=', product.product_tmpl_id.id), ('applied_on', '=', '1_product'),
-                '&', ('product_id', '=', product.id), ('applied_on', '=', '0_product_variant')]
+            domain = [
+                '&',
+                    '|',
+                        '&', ('product_tmpl_id', '=', product.product_tmpl_id.id), ('applied_on', '=', '1_product'),
+                        '&', ('product_id', '=', product.id), ('applied_on', '=', '0_product_variant'),
+                    ('pricelist_id.active', '=', True)
```

Comment:
```suggestion
                    ('pricelist_id.active', '=', True)
                    '|',
                        '&', ('product_tmpl_id', '=', product.product_tmpl_id.id), ('applied_on', '=', '1_product'),
                        '&', ('product_id', '=', product.id), ('applied_on', '=', '0_product_variant'),
```
We can drop one level of indentation if we rely on the fact that by default domain parts are aggregated with `&`.